goas/cl/cl/tests/simple_transfer.rs

```diff
@@ -0,0 +1,41 @@
+use rand_core::CryptoRngCore;
+
+fn receive_utxo(
+    note: cl::NoteWitness,
+    nf_pk: cl::NullifierCommitment,
+    rng: impl CryptoRngCore,
+) -> cl::OutputWitness {
+    cl::OutputWitness::random(note, nf_pk, rng)
+}
+
+#[test]
+fn test_simple_transfer() {
+    let mut rng = rand::thread_rng();
+
+    let sender_nf_sk = cl::NullifierSecret::random(&mut rng);
+    let sender_nf_pk = sender_nf_sk.commit();
+
+    let recipient_nf_pk = cl::NullifierSecret::random(&mut rng).commit();
+
+    // Assume the sender has received an unspent output from somewhere
+    let utxo = receive_utxo(cl::NoteWitness::basic(10, "NMO"), sender_nf_pk, &mut rng);
+
+    // and wants to send 8 NMO to some recipient and return 2 NMO to itself.
+    let recipient_output =
+        cl::OutputWitness::random(cl::NoteWitness::basic(8, "NMO"), recipient_nf_pk, &mut rng);
+    let change_output =
+        cl::OutputWitness::random(cl::NoteWitness::basic(2, "NMO"), sender_nf_pk, &mut rng);
+
+    let ptx_witness = cl::PartialTxWitness {
+        inputs: vec![cl::InputWitness::random(utxo, sender_nf_sk, &mut rng)],
+        outputs: vec![recipient_output, change_output],
+    };
+
+    let ptx = ptx_witness.commit();
+
+    let bundle = cl::Bundle {
+        partials: vec![ptx],
+    };
+
+    assert!(bundle.is_balanced(ptx_witness.balance_blinding()))
```

Comment:
couldn't we ignore blindings? This can be checked only if you know the inputs, at which point you can just look at the unit/values

Reply:
so, in the bundle proof, we use a SNARK to prove knowledge of some blinding s.t.

    bundle.balance = balance_cm(0, G, blinding)

---

Maybe there's a way to avoid this SNARK prove if we are clever with our blindings?

I believe monero does something like this.

Reply:
I mean, for risc0 we could just prove code that verifies units/value match and it would probably be faster, but I guess this is to support other implementations?

Reply:
yeah, that's true 🤔